File: List.cpp
#include<iostream>
using namespace std;

static int end=-1;
static int *p=new int[2];
static int length=2; 	
	 	
template <class T>
T Resize(T n1)
{
	cout<<"Older length: "<<length<<endl;
	int *p1=new int[2*length];
	//cout<<sizeof(p1);///here sizeof not working it is only showing the value 8 at every condition
	length=2*length;
	cout<<"New length"<<(length)<<endl;
	
	for(int temp=0;temp<length;temp++)
	{
		p1[temp]=0;
		//p1[temp]=p[temp];
	}
	for(int temp=0;temp<=end;temp++)
	{
		p1[temp]=p[temp];
	}
	//cout<<"New Array: ";
	/*for(int temp=0;temp<length;temp++)
	{
		cout<<" "<<p1[temp];
	}*/
	p=p1;
	//cout<<"New Array1: ";
	/*for(int temp=0;temp<length;temp++)
	{
		cout<<" "<<*(p+temp);
	}*/
}
template <class T0>
T0 Add(T0 n1)
{
	if((end+1)==length)
	{
		Resize(0);
	}	
	//int temp=end;
	p[end+1]=n1;
	end++;
	for(int temp1=0;temp1<length;temp1++)
	{
		cout<<p[temp1]<<" ";
	}
	cout<<endl;
	cout<<"value of end: "<<end<<endl;
	cout<<endl;
	cout<<"value of length: "<<length<<endl;
	
}
template <class T1 >
T1 Insert(T1 n1, T1 n2)
{
	if((end+1)==length)
	{
		Resize(0);
	}
	for(int temp=(length-1);temp>=n1;temp--)
	{
		p[temp]=p[temp-1];
	}
	p[n1]=n2;
	end++;
	for(int temp1=0;temp1<length;temp1++)
	{
		cout<<p[temp1]<<" ";
	}
	cout<<endl;
	cout<<"value of end: "<<end<<endl;
	cout<<endl;
	cout<<"value of length: "<<length<<endl;
	
}
template <class T2>
T2 Delete(T2 n1)
{
	for(int temp=n1;temp<(length-1);temp++)
	{
		p[temp]=p[temp+1];
	}
	p[end]=0;
	length--;
	for(int temp1=0;temp1<length;temp1++)
	{
		cout<<p[temp1]<<" ";
	}
	end--;
	cout<<endl;
	cout<<"value of end: "<<end<<endl;
	cout<<endl;
	cout<<"value of length: "<<length<<endl;
}
template <class T3>
T3 Read(T3 n1)
{
	cout<<"Element: "<<p[n1]<<endl;
	
}
int main()
{
	//int *p;
	//p=a;
	while(1)
	{
		//int a[]={5,6,7,8};
		//end=-1;
		int choice;
		cout<<"list implementation"<<endl;
		cout<<"1.Add elements to list"<<endl;
		cout<<"2.Insert elements to list"<<endl;
		cout<<"3.Delete elements to list"<<endl;
		cout<<"4.Read elements to list"<<endl;
		cout<<"5.Resize list"<<endl;
		cout<<"6.Read capacity to list"<<endl;
		cin>>choice;
		switch(choice)
		{
			case 1:
				int temp;
				cout<<"enter element"<<endl;
				cin>>temp;
				Add(temp);
				break;
			case 2:
				int temp1;
				cout<<"enter Index No";
				cin>>temp;
				cout<<"enter Element";
				cin>>temp1;
				Insert(temp,temp1);
				break;
			case 3:
				int temp2;
				cout<<"enter index to delete";
				cin>>temp2;
				Delete(temp2);		
				break;
			case 4:
				int temp3;
				cout<<"enter index to read";
				cin>>temp3;
				Read(temp3);
				break;
			case 5:
				Resize(0);
				break;
			case 6:
				cout<<"Capacity is: "<<length;
				break;
			
		}
		
	}
}

File: List.py
#!/usr/bin/python36

import math
import keyword
import os
import subprocess as sp

end=-1
p=[0 for i in range(3)]
length=2

def Resize():
	global end
	global length
	global p
	p1=[0 for i in range((2*length)+1)]
	length=length*2
	print("New Length: ",length)
	for temp in range(end+1):
		 p1[temp]=p[temp]
	p=p1
	input("Enter to continue")
	#os.system("clear")

def Add(n1):
	global end
	global length
	global p
	if((end+1)==length):
		Resize()
	p[end+1]=n1
	end=end+1
	for i in range(length):
		print(p[i],end=" ")
	print("\nValue of End: ",end)
	print("Value of Length: ",length)
	#os.system("clear")
	input("Enter to continue")

def Insert(n1,n2):
	global end
	global length
	global p
	#print("n1:",n1)
	#print("n2:",n2)
	if((end+1)==length):
		Resize()
	for i in range((length-1),(n1-1),-1):
		p[i]=p[i-1]
	p[n1]=n2
	end=end+1
	for i in range(length):
		print(p[i],end=" ")
	print("\nValue of End: ",end)
	print("\nValue of Length: ",length)
	#os.system("clear")
	input("Enter to continue")

def Delete(n1):
	global end
	global length
	global p
	for i in range(n1,(length-1),1):
		p[i]=p[i+1]
	p[end]=0
	length=length-1
	end=end-1
	for i in range(length):
		print(p[i],end="")
	print("\nValue of End: ",end)
	print("Value of Length: ",length)
	#os.system("clear")
	input("Enter to continue")

while True:
	print("""
	LIST IMPLEMENTATION:::
	1.Add element
	2.Insert element
	3.Delete element
	4.Resize array
	5.To Exit
	""")
	choice=int(input("enter value"))
 
	if choice==1:
		
		n1=int(input("enter element"))
		Add(n1)
		os.system("clear")

			
	elif choice==2:
		n2=int(input("enter index no"))
		n3=int(input("enter element"))
		Insert(n2,n3)
		os.system("clear")
			
	elif choice==3:
		n4=int(input("enter index to be deleted"))
		Delete(n4)
		os.system("clear")

	elif choice==4:
		Resize()
		os.system("clear")
				
	elif choice==5:
		exit()
	
	#input("Enter to continue...")
	#sp.getoutput("clear")	

	
	











